### ISSUE TITLE
Does the Code Support LoRA Fine-Tuning?

### ISSUE THREAD

**Issue by guijuzhejiang** [Robbyant/lingbot-vla-v2#27]
Hi, I noticed that lingbotvla/utils/lora_utils.py contains LoRA-related code, but I couldn't find where it is integrated into the training pipeline.

Does the current codebase support LoRA fine-tuning? If so, could you please provide an example or instructions on how to enable and run it?

Thanks!

**Comment by 724369308** [comment]
it does not support lora. you should add lora by yourself.

**Comment by guijuzhejiang** [comment]
Thanks . Since the model is quite large and does not fit in 48 GB of GPU memory during training, may I ask why LoRA fine-tuning is not supported? Is it because LoRA leads to significantly worse performance compared with full fine-tuning?

**Comment by 724369308** [comment]
My lora fine-tuning leads to worse performance, but I haven't found the reason since then.

**Comment by guijuzhejiang** [comment]
Hi, I tried adding LoRA for fine-tuning, but the loss becomes `NaN` from the beginning. Here is the log:

`Step 41/15780, Loss nan, VLA_Loss nan, Depth_Loss nan, Future_Depth_Loss nan, FutureVideo_Loss nan, SeqWise_Loss nan, RouterZ_Loss nan, MaxVio 7.0000, AvgSigmoid nan, GradNorm nan, LR 1.00e-05`

Are there any specific settings or modifications needed when using LoRA? Have you encountered a similar issue when adding LoRA?

**Comment by 724369308** [comment]
I haven't encountered this issue. My LoRA VLA loss can converge normally to the order of 6e‑2.

**Comment by LuFan31** [comment]
Sorry, we have not yet integrated LoRA post-training into our training pipeline.

Regarding the difficulty you reported running post-training with 48 GB of GPU memory, our code supports both **gradient checkpointing** and **gradient accumulation** to help reduce memory usage.

First, you can enable gradient checkpointing in your training config:

```yaml
train:
  enable_gradient_checkpointing: true
```

This reduces GPU memory usage at the cost of slower training.

For instructions on using gradient accumulation, please refer to the [Train — Batch Size & Gradient Accumulation](https://github.com/Robbyant/lingbot-vla-v2/blob/main/configs/vla/Training_Config.md#train--batch-size--gradient-accumulation) section in `configs/vla/Training_Config.md`.
